snappy/snapdata.go

```diff
@@ -45,6 +45,23 @@ func RemoveSnapData(snap *snap.Info) error {
 	return nil
 }
 
+// RemoveSnapSharedData removes the data shared between versions of the given snap
+func RemoveSnapSharedData(snap *snap.Info) error {
+	dirs, err := snapSharedDataDirs(snap)
+	if err != nil {
+		return err
+	}
+
+	for _, dir := range dirs {
+		if err := os.RemoveAll(dir); err != nil && !os.IsNotExist(err) {
+			return err
+		}
+		os.Remove(filepath.Dir(dir))
```

Comment:
Done. Though note: the later `os.Remove()` call is what removes the parent directories when they're empty, so I've left that, but added a comment to clarify.